lib/declare_schema/model/field_spec.rb

```diff
@@ -60,18 +60,18 @@ def initialize(model, name, type, position: 0, **options)
         @position = position
         @options = options.dup
 
-        @options.has_key?(:null) or @options[:null] = false
+        @options.has_key?(:null) or @options[:null] = Generators::DeclareSchema::Migration::Migrator.default_null
 
         case type
         when :text
           if self.class.mysql_text_limits?
             @options[:default].nil? or raise MysqlTextMayNotHaveDefault, "when using MySQL, non-nil default may not be given for :text field #{model}##{@name}"
-            @options[:limit] = self.class.round_up_mysql_text_limit(@options[:limit] || MYSQL_LONGTEXT_LIMIT)
+            @options[:limit] = self.class.round_up_mysql_text_limit(@options[:limit] || Generators::DeclareSchema::Migration::Migrator.default_text_limit)
           else
             @options[:limit] = nil
           end
         when :string
-          @options[:limit] or raise "limit: must be given for :string field #{model}##{@name}: #{@options.inspect}; do you want `limit: 255`?"
+          @options[:limit] ||= Generators::DeclareSchema::Migration::Migrator.default_string_limit or raise "limit: must be given for :string field #{model}##{@name}: #{@options.inspect}; do you want `limit: 255`?"
```

Comment:
A little confused on allowing default `nil`. Should we allow default to be `nil` for 2-4 mentioned in the [ticket](https://ringrevenue.atlassian.net/browse/TECH-5115) and raise if not explicitly set in field spec declaration.

Reply:
Right, for each of those, if it has a `nil` default, raise an exception if it's not explicitly set in the the field spec options, saying a value needs to be provided (because the default is configured `nil`).

Reply:
This last exception should also mention the config variable to tune.